src/utils/formatDateLocalized.ts

```diff
@@ -0,0 +1,30 @@
+import { format } from 'date-fns';
+
+/**
+ * Formats a date with localized pattern and ensures days are zero-padded.
+ * Uses date-fns format tokens for localization while guaranteeing consistent day padding.
+ *
+ * @param date - The date to format (Date object, timestamp, or ISO string)
+ * @param formatString - date-fns format string (e.g., 'PP', 'PP p')
+ * @returns Formatted date string with zero-padded days
+ *
+ * @example
+ * formatDateLocalized(new Date('2025-10-03'), 'PP')
+ * // English: "Oct 03, 2025"
+ * // French: "03 oct. 2025"
+ * // German: "03.10.2025"
+ */
+export const formatDateLocalized = (
```

Comment:
I think that you can use `dd` to have the padded 0 already, but you then have to build the whole PP yourself :s